lib/domain/timeline/watch_timeline_syncer.dart

```diff
@@ -15,29 +15,30 @@ class WatchTimelineSyncer {
 
   Future<int> syncPinDatabaseWithWatch() async {
     try {
-      final pins = await timelinePinDao.getAllPinsWithPendingSyncAction();
-      for (final pinToSync in pins) {
-        if (pinToSync.nextSyncAction == NextSyncAction.Upload) {
-          final res = await timelineControl.addPin(pinToSync.toPigeon());
+      final pinsToDelete = await timelinePinDao.getAllPinsWithPendingDelete();
+      for (final pinToDelete in pinsToDelete) {
```

Comment:
Can you do this in parallel? 
```
await Future.wait(pinsToDelete.map((pinToDelete) => [...]timelinePinDao.delete(pinToDelete.itemId)));
```

Reply:
What would that achieve? Database would do it serially regardless, it would just use internal locking.

Reply:
Fair point